### PR TITLE
fix codeql warnings

### PR DESCRIPTION
**Pull request by strega-nil-ms**
Link to internal codeql bugs:

- VSO-1746341
- VSO-1746390

I'm still not sure if we should have these, and I'm unsure how to test these changes.

### CODE REVIEW

File: stl/inc/charconv [hunk 1]
```diff
@@ -448,7 +448,9 @@ _NODISCARD inline uint32_t _Bit_scan_reverse(const _Big_integer_flt& _Xval) noex
 
     unsigned long _Index; // Intentionally uninitialized for better codegen
 
-    _BitScanReverse(&_Index, _Xval._Mydata[_Bx]); // assumes _Xval._Mydata[_Bx] != 0
+    // assumes _Xval._Mydata[_Bx] != 0
+    _BitScanReverse(
+        &_Index, _Xval._Mydata[_Bx]); // lgtm [cpp/conditionallyuninitializedvariable] _Xval._Mydata[_Bx] != 0
```
Comment (AlexGuteniev) [inline stl/inc/charconv:453]: If using `__assume` works, I would rather do this. 
Or ditto if the following works:
```C++
auto blah = _Xval._Mydata[_Bx];
if (blah != 0) {
    _BitScanReverse(
        &_Index, _Xval._Mydata[_Bx]);
} else {
    _Unreachable();
}
```
Sure by _works_ I imply not making codegen worse and suppressing `codeql` warnings.

**Review by AlexGuteniev — COMMENTED** [review]
Have you tried more documented and common annotations, instead of some specific to a particular tool (I first time read about)?

File: stl/inc/charconv [hunk 2]
```diff
@@ -448,7 +448,9 @@ _NODISCARD inline uint32_t _Bit_scan_reverse(const _Big_integer_flt& _Xval) noex
 
     unsigned long _Index; // Intentionally uninitialized for better codegen
 
-    _BitScanReverse(&_Index, _Xval._Mydata[_Bx]); // assumes _Xval._Mydata[_Bx] != 0
+    // assumes _Xval._Mydata[_Bx] != 0
```
Comment (CaseyCarter) [inline stl/inc/charconv:451]: I don't think we need to repeat this here, on 447, and on 453. I suggest moving the `_STL_INTERNAL_CHECK` down to immediately precede the call to `_BitScanReverse`, and removing the other two occurrences.

File: stl/inc/system_error [hunk 3]
```diff
@@ -579,7 +579,8 @@ public:
             constexpr size_t _Unknown_error_length = sizeof(_Unknown_error) - 1; // TRANSITION, DevCom-906503
             return string{_Unknown_error, _Unknown_error_length};
         } else {
-            return string{_Msg._Str, _Msg._Length};
+            return string{_Msg._Str,
+                _Msg._Length}; // lgtm [cpp/uninitializedptrfield] _Msg._Length != 0 \implies _Msg._Str is initialized
```
Comment (CaseyCarter) [inline stl/inc/system_error:583]:
```suggestion
                _Msg._Length}; // lgtm [cpp/uninitializedptrfield] (_Msg._Length != 0) == (_Msg._Str != nullptr)
```
Feel free to `_STL_INTERNAL_CHECK` the precondition and remove it from the suppression comment if you like.

**Pull request closed without merging**